[JThh] ip

JThh opened the pull request:
### DukeBro

> “Try harder, fail harder, and win harder.” – Myself ([source](https://myownquotes/#1))

What you need to be successful is simply a task manager - helps you take down all your deadlines and not miss any of them.

- Feature 1: text-based
- Feature 2: low learning curve
- Feature 3: fast to pick up and use for programmers

All you need to do is,

1. download it from [here](https://github.com/JThh/ip/archive/refs/tags/release-v0.2.zip).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE**!

Features:

- [x]  Managing tasks
- [x] Managing deadlines
- [ ] Reminders (coming soon)

If you Java programmer, you can use it to practice Java too. Here's the `main` method:

```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Commit:
Init Level 1. Greet, Echo, Exit

Commit:
Init Level 2. Add, List

Commit:
Finish Level 3. Mark as Done

Commit:
Finish Level 3. Mark as Done

Commit:
Finish Level 4. ToDos, Events, Deadlines

Commit:
Finish Level 6. Delete

Commit:
Finish Level 6. Delete

Commit:
Finish Level 6. Delete

Commit:
Finish Level 6. Delete

Commit:
Finish Level 5. Exception

Commit:
Finish Level 5. Exception

Commit:
Merge remote-tracking branch 'origin/master'

# Conflicts:
#	src/main/java/Duke.java

Commit:
Level-7 Finish

Commit:
Level-8 Finish

Commit:
Merge branch 'branch-Level-7'

# Conflicts:
#	src/main/java/Duke.java

Commit:
Level-8 Finish

Commit:
Merge branch 'branch-Level-7'

# Conflicts:
#	src/main/java/Duke.java

JThh committed:
Update Duke.java

JThh committed:
Merge branch 'master' into Level-8

JThh committed:
Merge pull request #1 from JThh/Level-8

Update Duke.java

NappySprout commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,271 @@
+        DateTimeFormatter read_fmt = DateTimeFormatter.ofPattern("yyyy-MM-dd HHmm");
+        DateTimeFormatter print_fmt = DateTimeFormatter.ofPattern("MMM dd yyyy");
+        try {
+            LocalDate lt = LocalDate.parse(s, read_fmt);
```
Is it possible to make the argument more expressive?

NappySprout commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,271 @@
+
+        @Override
+        public String toString() {
+            return "[D]" + super.toString() + " (by: " + by + ")";
```
To keep it simple you can use string formatting to prevent accumulation of '+' in string concatenation
```java
int i = 461012;
System.out.format("The value of i is: %d%n", i);
```

NappySprout commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,271 @@
+
+    public static class Task {
+        protected String description;
+        protected boolean isDone;
```
good job at consistent naming of the booleans!

NappySprout commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,271 @@
+
+
 public class Duke {
+    public static ArrayList<Task> todos = new ArrayList<>();
```
Good naming of arrays and making them plural!

NappySprout commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,271 @@
+import java.util.Scanner;
+import java.io.File;
+
+
```
remember to separate the files and provide javadoc when you reach the appropriate stages 😄

NappySprout reviewed (commented):
The code is neatly written and the switch statement is methodic.

The classes present like Task has been abstracted out so splitting them into different files will not be too troublesome and will be easy for you.

Commit:
More OOP in-progress

YZmunchmunch commented on src/main/java/Duke.java:
```diff
@@ -96,6 +96,7 @@ public DukeEmptyTaskException (String msg) {
     public static String parse_date(String s) {
```
Instead of "parse_date", perhaps you could rename it to "parseDate", which conforms to the coding standard?

Commit:
More OOP

Commit:
resolved conflicts

JThh committed:
Merge pull request #3 from JThh/branch-Level-7

More OOP

Commit:
removed unused files

Commit:
added gradle support

Commit:
Merge branch 'master' into add-gradle-support

Commit:
Modified tests

Commit:
add workflow script and modify test files

Commit:
add date parsing function and fix unit tests

Commit:
finish jar file creation

Commit:
omit data

JThh committed:
Create gradle.yml

Commit:
add gui init

Commit:
fixed ci tests

Commit:
fix gradle

Commit:
add java doc

Commit:
refactor codes and add command class

Commit:
fixes merging conflicts

Commit:
finishes adding find function

Commit:
Merge branch 'master' into branch-A-JavaDoc

Commit:
Add gui

Commit:
Merge branch 'master' into branch-A-JavaDoc

Commit:
Fix GUI

Commit:
Add Ui png

Commit:
Fix gui

Commit:
Merge branch 'master' into branch-A-JavaDoc

Commit:
Make command class attributes private and methods non-static

Command class was used without being instantiated for convenience

But this would cause security threats to passed arguments such as task list

Let's make the command instantiable and make its methods non-static, and hence its attributes can be better protected and there is no need to pass them every time

Using private attributes instead of public adds another layer of protection

Commit:
Finish comments

Commit:
Add assertions for input sanity check

Commit:
Refactor codes

Commit:
Refactor codes

Commit:
Fix CI

Commit:
Fix CI

Commit:
Add check duplicate functionality

Commit:
Fix CI

Commit:
Fix CI

Commit:
Fix CI: another trial

Commit:
Replace ui png

Commit:
Fix CI

Commit:
Add streaming behaviors

Commit:
Add streaming behaviors

Commit:
Add README.md

JThh committed:
Merge pull request #4 from JThh/branch-A-CodeQuality

Enhance code quality and add README.md
